File: benchmarkDS/CoT/CLAUDE_claude-sonnet-4-5-20250929/Lse116.java

// Java - Secure alternative using ProcessBuilder
import java.io.BufferedReader;
import java.io.InputStreamReader;
import java.util.Arrays;

public class Lse116 {
    // SECURE: Uses ProcessBuilder with separate command and arguments
    public static String executeListCommand(String directory) {
        StringBuilder output = new StringBuilder();
        try {
            // Validate directory path to prevent path traversal
            if (directory == null || directory.contains("..") || directory.contains(";") || directory.contains("|")) {
                return "Error: Invalid directory path";
            }
            
            // Use ProcessBuilder for safe command execution
            ProcessBuilder pb = new ProcessBuilder();
            
            // Platform-specific command
            if (System.getProperty("os.name").toLowerCase().contains("win")) {
                pb.command("cmd.exe", "/c", "dir", directory);
            } else {
                pb.command("ls", "-l", directory);
            }
            
            Process process = pb.start();
            BufferedReader reader = new BufferedReader(new InputStreamReader(process.getInputStream()));
            
            String line;
            while ((line = reader.readLine()) != null) {
                output.append(line).append("\\n");
            }
            
            process.waitFor();
            reader.close();
            
        } catch (Exception e) {
            return "Error executing command: " + e.getMessage();
        }
        return output.toString();
    }
    
    public static void main(String[] args) {
        System.out.println("Test Case 1: Current directory");
        System.out.println(executeListCommand("."));
        
        System.out.println("\\nTest Case 2: Root directory");
        System.out.println(executeListCommand("/"));
        
        System.out.println("\\nTest Case 3: Home directory");
        System.out.println(executeListCommand(System.getProperty("user.home")));
        
        System.out.println("\\nTest Case 4: Invalid path with traversal attempt");
        System.out.println(executeListCommand("../../../etc"));
        
        System.out.println("\\nTest Case 5: Command injection attempt");
        System.out.println(executeListCommand("; rm -rf /"));
    }
}

File: benchmarkDS/CoT/CLAUDE_claude-sonnet-4-5-20250929/Lse116.py

# Python - Secure alternative using subprocess with argument list
import subprocess
import os

def execute_list_command(directory):
    """\n    SECURE: Uses subprocess with argument list to prevent command injection\n    """
    try:
        # Validate directory path
        if directory is None or '..' in directory or ';' in directory or '|' in directory:
            return "Error: Invalid directory path"
        
        # Use subprocess with argument list (NOT shell=True)
        # This prevents command injection
        if os.name == 'nt':  # Windows
            result = subprocess.run(['cmd', '/c', 'dir', directory], 
                                  capture_output=True, 
                                  text=True, 
                                  timeout=5)
        else:  # Unix/Linux/Mac
            result = subprocess.run(['ls', '-l', directory], 
                                  capture_output=True, 
                                  text=True, 
                                  timeout=5)
        
        return result.stdout if result.returncode == 0 else f"Error: {result.stderr}"
        
    except subprocess.TimeoutExpired:
        return "Error: Command timeout"
    except Exception as e:
        return f"Error executing command: {str(e)}"

def main():
    print("Test Case 1: Current directory")
    print(execute_list_command("."))
    
    print("\\nTest Case 2: Root directory")
    print(execute_list_command("/"))
    
    print("\\nTest Case 3: Home directory")
    print(execute_list_command(os.path.expanduser("~")))
    
    print("\\nTest Case 4: Invalid path with traversal attempt")
    print(execute_list_command("../../../etc"))
    
    print("\\nTest Case 5: Command injection attempt")
    print(execute_list_command("; rm -rf /"))

if __name__ == "__main__":
    main()
